### ISSUE TITLE
How it's possible to call a custom confirmation BEFORE default confirmation?

### ISSUE THREAD

**Issue by evje** [volkanceylan/Serenity#2994]
Hi, Volkan Ceylan and all Serenity community!
I continue to learn great Serenity framework. [Here](https://github.com/volkanceylan/Serene/tree/master/Serene/Serene.Web/Modules/BasicSamples/Grids/CancellableBulkAction) I find an example of CancellableBulkAction. It looks as at screenshot

![example](https://user-images.githubusercontent.com/16364615/33473358-36514bea-d687-11e7-85dd-a5db1bac9b52.png)

But, it's possible a case, when we need some custom check before final confirmation. To implement that feature, I added to [CancellableBulkActionGrid.ts](https://github.com/volkanceylan/Serene/blob/master/Serene/Serene.Web/Modules/BasicSamples/Grids/CancellableBulkAction/CancellableBulkActionGrid.ts) such code:

```
protected getButtons() {
            return [{
                title: 'Perform Bulk Action on Selected Orders',
                cssClass: 'send-button',
                onClick: () => {
                    if (!this.onViewSubmit()) {
                        return;
                    }

                    this.showCustomWarning("Warning").done(() => {
                        // they pressed OK
                        alert("Yes");
                    }).fail(() => {
                        // the pressed Cancel
                        alert("No");
                    });

                    var action = new OrderBulkAction();
                    action.done = () => this.rowSelection.resetCheckedAndRefresh();
                    action.execute(this.rowSelection.getSelectedKeys());
                }
            }];
        }
```

```
private showCustomWarning(title: string): JQueryPromise<Object> {
            var def = $.Deferred();

            $('<div class="ui-dialog ui-corner-all ui-widget ui-widget-content ui-front s-MessageDialog s-WarningDialog ui-dialog-buttons ui-draggable">' +
                '<div class="ui-dialog-content ui-widget-content">' +
                '<div class="message">Do you want to continue?' +
                '</div></div></div>').dialog({
                modal: true,
                title: title,
                buttons: {
                    Yes: function () {
                        def.resolve();
                        $(this).dialog("close");
                    },
                    No: function () {
                        def.reject();
                        $(this).dialog("close");
                    }
                }
            });

            return def.promise();
        }
```

But I catch unexpected behavior, see at screenshot:

![needed](https://user-images.githubusercontent.com/16364615/33473603-29518a4e-d688-11e7-8943-76e7b37fa3bc.png)

So, the default confirm window appears firstly and has greater priority than my custom confirm window, and I can't operate my own confirm window until the default confirm window is open. But I need, that my window appears firstly and only after clicking Yes or No the default window comes on.

**Please, give me an advise, what should I do to implement this idea?**

**Comment by Zahar661** [comment]
![image](https://user-images.githubusercontent.com/26430136/33474661-a5c90a6c-d68c-11e7-8f23-a90c77f6f65b.png)
![image](https://user-images.githubusercontent.com/26430136/33474717-d326afa0-d68c-11e7-96ab-16262f0d9c61.png)
![image](https://user-images.githubusercontent.com/26430136/33474775-01c231d6-d68d-11e7-8abb-bf43980ece07.png)

**Comment by Zahar661** [comment]
?
![image](https://user-images.githubusercontent.com/26430136/33474908-716fe4a6-d68d-11e7-8759-7f622183a60e.png)

**Comment by volkanceylan** [comment]
Nice presentation @Zahar661 :)

**Comment by evje** [comment]
@Zahar661 , great job!
Thank U very much)

Also it's possible such code:

```
protected getButtons() {
            return [{
                title: 'Perform Bulk Action on Selected Orders',
                cssClass: 'send-button',
                onClick: () => {
                    if (!this.onViewSubmit()) {
                        return;
                    }

                    Q.confirm(
                        "Do you want to continue?",
                        () => {
                            //Q.notifySuccess("Allright!");
                            var action = new OrderBulkAction();
                            action.done = () => this.rowSelection.resetCheckedAndRefresh();
                            action.execute(this.rowSelection.getSelectedKeys());
                        },
                        {
                            onNo: () => {
                                //Q.notifyInfo("You clicked NO. Why?");
                            },
                            onCancel: () => {
                                //Q.notifyError("You clicked X. Operation is cancelled. Will try again?");
                            },
                            title: "Some Custom Confirmation Title"
                        });
                }
            }];
        }
```

**Comment by danpadua** [comment]
Perfect +1
